x-pack/plugins/security_solution/public/resolver/view/panel.tsx

```diff
@@ -205,21 +197,12 @@ const PanelContent = memo(function PanelContent() {
     return 'processListWithCounts';
   }, [uiSelectedEvent, crumbEvent, crumbId, graphableProcessEntityIds]);
 
-  useEffect(() => {
-    // dispatch `appDisplayedDifferentPanel` to sync state with which panel gets displayed
-    dispatch({
-      type: 'appDisplayedDifferentPanel',
-      payload: panelToShow,
-    });
-  }, [panelToShow, dispatch]);
```

Comment:
:grey_question: So even though `panelToShow` has a dep on `crumbEvent,crumbId` -> which come from the `useHistory` hook (which should change when the URL changes?) it _doesn't_ trigger this effect?

Reply:
disregard any stuff I said in chat. my original assumptions were wrong. this does trigger, but on the render that triggers, the panel still renders using the old `currentPanelView` value from State. that is fine except that those components get props which are (in part) based on data from useLocation. so it's like half the data is at time `t` and half is at time `t + 1`

Reply:
![triage mov](https://user-images.githubusercontent.com/35559/86976587-ce3afd80-c148-11ea-8aee-438b7d0a11d4.gif)
This gif shows me debugging the issue on `master`. First I added this code above the `return` in `panel.tsx`

```
  console.log(
    'panelToShow',
    panelToShow,
    'currentPanelView',
    currentPanelView,
    'relatedStatsForIdFromParams',
    relatedStatsForIdFromParams
  );
```

The goal is to show that `panelToShow` and `currentPanelView` are out of sync, and that other values which are used as props to panels aren't in sync with `currentPanelView` because they are (in part) based on `useHistory` (or `useLocation`, or anything else not in `redux`.) Normally redux makes all state transitions atomic and ensures that all pieces of state are in sync. 

In the gif, when the error occurs I scroll up and you can see this:
![image](https://user-images.githubusercontent.com/35559/86976712-0d694e80-c149-11ea-92a1-7224292ed28e.png)

`currentPanelView` is `eventCountsForProcess`. That renders the `EventCountsForProcess` component which takes `relatedStatsForIdFromParams`. In fact the code assumes that `relatedStatsForIdFromParams` will be truthy if `currentPanelView` is `eventCountsForProcess`. You can see that `relatedStatsForIdFromParams` is actually undefined. It is based (in part) on `idFromParams` which is in turn based on `queryParams` which is based on `useHistory` (or `useLocation`.)

Data from `useHistory` (or `useLocation`) will be up to date at the time of render. But `currentPanelView` is behind because this component dispatches and event which changes it whenever `panelToView` changes. 

The solution in the PR isn't a good long term solution. Values calculated in the render method of this component use data from redux and data thats not from redux. It also updates the redux data (via dispatch) based on data not from redux. Without a single, atomically modified, state container that is shared across all components, keeping this stuff in sync will be tricky.

 It is passed with the non-null assertion operator, so typescript doesn't catch the fact that it's undefined.

Reply:
Initially, you asked for that so that we at least had some valence with which panel was showing in state. Is there a way to synchronize it / make it work so it doesn't twitch when the URL changes?